### ISSUE TITLE
unable to use train.run() more than 2 times in row (yolov5)

### ISSUE THREAD

**Issue by ShAmoNiA** [ultralytics/yolov5#9737]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Training

### Bug

Greeting,

I just want to use **run** function in train.py which located in yolov5 folder like below:

```
train_info = train.run(data='*', weights='*',epochs=*, batch_size=*)
```

it works find if I just use it once in my code, however in if I use it more than 2 times, my code will hang.

```
train_info = train.run(data='yolov5/data/test1.yaml', weights='yolov5n.pt',epochs=1, batch_size=8)
train_info = train.run(data='yolov5/data/test2.yaml', weights='yolov5n.pt',epochs=1, batch_size=8)
train_info = train.run(data='yolov5/data/test3.yaml', weights='yolov5n.pt',epochs=1, batch_size=8)
``` 

I faced with a problem into yolov5 when I tried to train with upper form.
if we assume that the result for first and second train will be saved in exp17 and exp18, in third train, I saw this:

[issue][1]

why exp17/labels.jpg and exp18/labels.jpg were loaded here?

that thing happened too when second train was running. I could see exp17/labels.jpg was loaded there!

any suggestion for fixing it?

  [1]: https://i.stack.imgur.com/ZpnNa.png

### Environment

Yolov5
torch-1.10.1+cu102 CUDA:0 (NVIDIA GeForce RTX 2080 Ti, 11017MiB)
CentOS Linux 7 (Core) - Linux 3.10.0-1127.el7.x86_64
Python-3.6.8 

### Minimal Reproducible Example

```
from yolov5 import train

train_info = train.run(data='yolov5/data/test.yaml', weights='yolov5n.pt',epochs=1, batch_size=8)
train_info = train.run(data='yolov5/data/test.yaml', weights='yolov5n.pt',epochs=1, batch_size=8)
train_info = train.run(data='yolov5/data/test.yaml', weights='yolov5n.pt',epochs=1, batch_size=8)

```

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by ShAmoNiA** [comment]
this is log info for upper code

[output.txt](https://github.com/ultralytics/yolov5/files/9741399/output.txt)

**Comment by glenn-jocher** [comment]
@ShAmoNiA hi there! Thanks for reaching out. I'd recommend using the YOLOv5 `train` command just once in your script to avoid potential conflicts. You can always manipulate the training settings within the same `train` command multiple times by adjusting the parameters. This approach will prevent any unexpected behavior or conflicts with the training process.

In the meantime, could you please provide the output.txt directly in the issue description? That will allow us to take a closer look at the specific error messages and help you more effectively. 

Thanks for your understanding!
